Java

```diff
@@ -1,44 +1,55 @@
 import java.util.Scanner; 
-import java.text.NumberFormat;
-class student{   
+    class Student{
+        String name;
+        String id;
+        int hours = 0;
+
+        public Student(String name, String id){
+            this.name = name;
+            this.id = id;
+        }
+
+        public void addHours(double hours){
+            this.hours += hours;
+        }
+
+        public double getHoursAsPercentage(double totalHours){
+            return hours * 100.0 / totalHours;
+        }
+    }
+class Main{   
 	
 	public static void main(String args[]) { 
 	
 		Scanner s = new Scanner(System.in); 
 		String newline = System.lineSeparator(); 
-		NumberFormat defaultFormat = NumberFormat.getPercentInstance();
-		defaultFormat.setMinimumFractionDigits(2);
```

Comment:
Is declared but not used (could have been at line 74: `System.out.println("Overall attendance is : " + defaultFormat.format(perc) + "%"));`)